miniapps/navier/navier_shear.cpp

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2010-2020, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#include "navier_solver.hpp"
+#include <fstream>
+
+using namespace mfem;
+using namespace navier;
+
+struct s_NavierContext
+{
+   int order = 6;
+   double kinvis = 1.0 / 100000.0;
+   double t_final = 10 * 1e-3;
+   double dt = 1e-3;
+} ctx;
+
+void vel_shear_ic(const Vector &x, double t, Vector &u)
+{
+   double xi = x(0);
+   double yi = x(1);
+
+   double rho = 30.0;
+   double delta = 0.05;
+
+   if (yi <= 0.5)
+   {
+      u(0) = tanh(rho * (yi - 0.25));
+   }
+   else
+   {
+      u(0) = tanh(rho * (0.75 - yi));
+   }
+
+   u(1) = delta * sin(2.0 * M_PI * xi);
+}
+
+int main(int argc, char *argv[])
+{
+   MPI_Session mpi(argc, argv);
+
```

Comment:
No command line options?

Reply:
This is a very specific and sensitive test case, so I'd rather not provide options which tend to break it.